app/controllers/jobs/new/build-answer-set.js

```diff
@@ -52,7 +52,10 @@ export default Ember.Controller.extend({
       }).then((savedStageGroup) => {
         return Ember.RSVP.all(
           // save the dds files
-          savedStageGroup.get('ddsFiles').map(ddsFile => { return ddsFile.save(); })
+          savedStageGroup.get('ddsFiles').map((ddsFile,index) => {
```

Comment:
>Is there any guarantee that this relationship is ordered in the way we expect?
>I think it would be safer to number the sequence in answerChanged. That's where the stage group gets set, and inputFiles is still an ordered array.

I don't mind moving this set method further up the chain.

Is there something dangerous about ember where it sometimes scrambles array data?

Reply:
> Is there something dangerous about ember where it sometimes scrambles array data?

No, but the [property you're enumerating](https://github.com/Duke-GCB/bespin-ui/blob/5056cf489a9d319b1c1eef27f710ae79232d3c81/app/models/job-file-stage-group.js#L4) (is an Ember Data `hasMany` relationship, which is [computed by ember data](https://github.com/emberjs/data/blob/0999f8a0a756a3118f1e6a4e110709be72e1ad3e/addon/-private/system/relationships/has-many.js#L147) and not necessarily the array you provided (it may not even be an array).The [documentation](https://guides.emberjs.com/v2.17.0/models/relationships/#toc_one-to-many) for hasMany relationships makes no mention of ordering of the elements and I couldn't find anything in the source code or tests to suggest that an order should be preserved.

While it might be a happy accident that it works out, I think it's a mistake to rely on that when we can easily move the line of code that sets the order to a place where we know what it should be.

Reply:
Discussed this with @dleehr. We want to handle the case where users pick multiple groups of files and retain the order displayed on the screen. To this end we will add a sequenceGroup to DDSJobInputFile/URLJobInputFile.